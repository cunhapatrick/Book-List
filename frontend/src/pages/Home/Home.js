import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types';
import Paginate from 'react-paginate';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

// Components
import Topbar from 'components/Topbar';
import Filterbar from 'components/Filterbar';
import List from 'components/List';
import Modal from 'components/Modal';

// Store
import * as BookActions from 'store/modules/books/actions';

// CSS
import { Container } from './Home.style';

const Home = ({ searchBooksRequest, loading, data }) => {
  const [filter, setFilter] = useState('');
  const [iData, setIData] = useState(1998);
  const [eData, setEData] = useState(2019);
  const [modal, setModal] = useState(false);
  const [modalData, setModalData] = useState({});

  const validationYear = year => {
    if (isNaN(year) || year < 0) return 1998;
    
    return year;
  }
	
	const handleFilter = ({ target }) => setFilter(target.value);
  const handleIData = ({ target }) => setIData(validationYear(target.value));
  const handleEData = ({ target }) => setEData(validationYear(target.value));
  
  const handleDetalhes = (book = {}) => {
    if (book._id) setModalData(book);
		setModal(!modal);
  }

	const handleSearch = () => {
		searchBooksRequest({
			filter,
			iData,
			eData,
		});
	}

	useEffect(() => {
		searchBooksRequest();
	}, [searchBooksRequest]);		
  return (
  <Container>
    {modal && <Modal 
			show={modal}
			data={modalData}
			closeModal={handleDetalhes}
		/>}
    <Topbar
      filter={filter}
			handleFilter={handleFilter}
			handleSearch={handleSearch}
			loading={loading}
    />
		<Filterbar
			iData={iData}
			eData={eData}
			handleIData={handleIData}
			handleEData={handleEData}
			total={data.total}
		/>
		<List
			handleDetalhes={handleDetalhes}
			books={data.docs}
		/>
		<div>
			<Paginate
				pageCount={data.pages}
				pageRangeDisplayed={3}
				marginPagesDisplayed={2}
				onPageChange={current_page => handleSearch(current_page.selected +1)}
				previousLabel={'<'}
				nextLabel={'>'}
				activeClassName="active"
				initialPage={data.page - 1}
				disabledClassName="disabled"
			/>
		</div>
  </Container>
)};

Home.propTypes = {
	loading: PropTypes.bool.isRequired,
	data: PropTypes.object.isRequired,
	searchBooksRequest: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
	loading: state.books.loading,
	data: state.books.data,
});

const mapDispatchToProps = dispatch =>
	bindActionCreators(BookActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Home);