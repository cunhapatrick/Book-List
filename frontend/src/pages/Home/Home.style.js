import styled from 'styled-components';

export const Container = styled.div`
  div {
    ul {
      margin: 3% 0 0 45%;
      text-align: center;
      li {
        text-align: center;
        list-style-type: none;
        border: 1px solid black;
        text-align: center;
        margin-left: 10px;
        float: left;
        padding: 5px 10px;
        font-weight: 500;
        font-size: 16px;

        &:hover:not(.disabled) {
          cursor: pointer;
          background: #A4A4C4;
          color: white;
        }

        &.active {
          color: white;
          background: #0F5FCF;
        }
        &.disabled {
          cursor: not-allowed;
        }
      }
    }
  }
`;