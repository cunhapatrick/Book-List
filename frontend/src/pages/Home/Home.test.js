import React from 'react';
import { mount } from 'enzyme';
import Home from './Home';

import configureStore from 'redux-mock-store';
const mockStore = configureStore();

const initialState = {
  books: {
    data: {
      docs: [
        {
          _id: 1,
          title: 'teste',
        }
      ],
      total: 5,
    },
    loading: true,
  },
  searchBooksRequest: () => {},
};

let store;
let wrapper;

describe('Testing Home Component', () => {

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Home store={store} />,
    );
  })

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});