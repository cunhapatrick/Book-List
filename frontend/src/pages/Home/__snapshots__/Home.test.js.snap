// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component should render properly 1`] = `ReactWrapper {}`;
